libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1364,6 +1364,8 @@ void TypeChecker::endVisit(ExpressionStatement const& _statement)
 {
 	if (type(_statement.expression())->category() == Type::Category::RationalNumber)
 		if (!dynamic_cast<RationalNumberType const&>(*type(_statement.expression())).mobileType())
+		// Add Check here (?)
+		// An inadvertent conversion could be happening in RationalType::mobileType()
 			m_errorReporter.typeError(3757_error, _statement.expression().location(), "Invalid rational number.");
```

Comment:
This checks literal expressions that are the only thing in a statement. I.e.

```soldidity
1 + 2 * 3;
```

Such statements have no effect so I think there's no point warning about the conversion here.